swayfmt/src/utils/map/newline.rs

```diff
@@ -278,7 +278,11 @@ fn insert_after_span(
         format_newline_sequence(&newline_sequence, threshold)
     )?;
     let mut src_rope = Rope::from_str(formatted_code);
-    src_rope.insert(at, &sequence_string);
+
+    if src_rope.try_insert(at, &sequence_string).is_err() {
+        return Err(FormatterError::NewlineSequenceError);
+    }
```

Comment:
Also possible to write this in a single line, if you prefer:
```suggestion
    src_rope.try_insert(at, &sequence_string).map_err(|_| FormatterError::NewlineSequenceError)?;
```

Reply:
Ah yeah, that's what I wanted to do! :)